server/src/test/java/org/elasticsearch/action/StepListenerTests.java

```diff
@@ -92,7 +91,7 @@ public void testAbortOnFailure() throws Exception {
 
         if (failedStep == 1) {
             assertThat(expectThrows(RuntimeException.class, step1::result).getMessage(), equalTo("failed at step 1"));
-            assertThat(expectThrows(RuntimeException.class, step2::result).getMessage(), equalTo("step is not completed yet"));
```

Comment:
Related to my comment on `StepListener.result` - this looks like you did intend to change the contract of StepLIstener.result to make it illegal to call it if the `StepListener` is not done?

Reply:
Yes I did.